### PR TITLE
Update 4.1 SDW to use Fedora-39

### LINKED ISSUE #975: Support Fedora-39 (must do by 2024-05-14)
Fedora 39 templates are available: https://www.qubes-os.org/news/2024/02/13/fedora-39-templates-available-fedora-38-approaching-eol/. Following the issue structure of [the previous version bump](https://github.com/freedomofpress/securedrop-workstation/issues/892):

- [ ] Bump supported_fedora_version to fedora-39 and convert system-wide default dispvm to fedora-39-dvm
- [ ] Create sd-fedora-39-dvm with USB autoattach rules and use as template for sys-usb (unless we decide to revisit this behaviour)
- [ ] Update docs, including developers.securedrop.org (multi-machine setup on Qubes) docs and workstation.securedrop.org, to include correct fedora version

Related:
  - https://github.com/freedomofpress/securedrop-workstation/issues/882

### PR DESCRIPTION
**Pull request by zenmonkeykstop**
## Status

Ready for review 

## Description of Changes

Fixes #975 

- updates fedora templates in use by sys* VMs to fedora-39 variants.
- adds workstation CI config file
- updates yum-test key.

### Prep
Build an RPM from this branch with `make build-rpm`, and copy it to dom0 for use with test plans below.

### Fresh install (prodlike install)
Qubes 4.1.2 expected, please note hardware
#### Testing:
- [x] RPM installs successfully in dom0
- [x] with config in place, `sdw-admin --apply` completes successfully
- [x] fedora-39 wis installed as part of `--apply` run
- [x] sys-* AppVMs use fedora-39 versions of their expected templates
- [x] basic client functionality requiring network access (login, first sync) completed successfully.
- [x] basic client functionality requiring USB support (export, ~print~) completed successfully
 
### Upgrade from 0.10.0 (no fedora-39 templates preinstalled)
Qubes 4.1.2 expected, please note hardware
#### Testing: 
- [x] RPM installs succesfully via `sudo dnf install <name>` in dom0, replacing the existing 0.10.0 rpm
- [x] with config in place, `sdw-admin --apply` completes successfully
- [x] fedora-39 wis installed as part of `--apply` run
- [x] sys-* AppVMs use fedora-39 versions of their expected templates
- [x] basic client functionality requiring network access (login, first sync) completed successfully.
- [x] basic client functionality requiring USB support (export, <strike>print</strike>) completed successfully

### Upgrade from 0.10.0 (with fedora-39 templates preinstalled)
Qubes 4.1.2 expected, please note hardware
#### Testing: 
- [ ] RPM installs succesfully via `sudo dnf install <name>` in dom0, replacing the existing 0.10.0 rpm
- [ ] with config in place, `sdw-admin --apply` completes successfully
- [ ] fedora-39 wis installed as part of `--apply` run
- [ ] sys-* AppVMs use fedora-39 versions of their expected templates
- [ ] basic client functionality requiring network access (login, first sync) completed successfully.
- [ ] basic client functionality requiring USB support (export, print) completed successfully

### CODE REVIEW

**Comment by zenmonkeykstop** [comment]
The sd-ci-runner failure seems to be related to an issue with downloading the base template RPM - I can replicate on a Qubes 4.1 machine. The test signing key was recently rotated, it looks like the RPM was re-signed as well but for some reason qubes-dom0-update is barfing on it.

The prod template and key are unchanged however - the test plan refers to prod setups so it should be possible to do QA on that basis.

File: rpm-build/SPECS/securedrop-workstation-dom0-config.spec [hunk 1]
```diff
@@ -124,8 +124,8 @@ find /srv/salt -maxdepth 1 -type f -iname '*.top' \
     | xargs qubesctl top.enable > /dev/null
 
 # Force full run of all Salt states - uncomment in release branch
-mkdir -p /tmp/sdw-migrations
-touch /tmp/sdw-migrations/whonix-17-update
+# mkdir -p /tmp/sdw-migrations
+# touch /tmp/sdw-migrations/fedora-39-update
```
Comment (eloquence) [inline rpm-build/SPECS/securedrop-workstation-dom0-config.spec:128]: Intentionally commented out for now?

**Comment by eloquence** [comment]
(Taking fresh install scenario.)

**Comment by eloquence** [comment]
Q: Is this intentionally targeting the `release/0.10.0` branch (previous release)?

**Comment by zenmonkeykstop** [comment]
No, it should have been 0.11.0, good catch.

**Comment by eloquence** [comment]
All good on fresh install. Can't test print (no supported printer) but export works; printer dialog/check also comes up. Boxes checked directly in test plan.

**Comment by eloquence** [comment]
(Test plan seems to be duplicated in release ticket; have checked it off there as well.)

**Comment by eloquence** [comment]
(Per quick chat, treating SDW CI fail as unrelated to this PR which has been independently QA'd.)

**Pull request merged**